op-node/rollup/derive/engine_queue.go

```diff
@@ -212,27 +207,26 @@ func (eq *EngineQueue) Step(ctx context.Context) error {
 	if eq.needForkchoiceUpdate {
 		return eq.tryUpdateEngine(ctx)
 	}
-	if len(eq.safeAttributes) > 0 {
+	if eq.safeAttributes != nil {
 		return eq.tryNextSafeAttributes(ctx)
 	}
 	outOfData := false
-	if len(eq.safeAttributes) == 0 {
-		newOrigin := eq.prev.Origin()
-		// Check if the L2 unsafe head origin is consistent with the new origin
-		if err := eq.verifyNewL1Origin(ctx, newOrigin); err != nil {
-			return err
-		}
-		eq.origin = newOrigin
-		eq.postProcessSafeL2() // make sure we track the last L2 safe head for every new L1 block
-		if next, err := eq.prev.NextAttributes(ctx, eq.safeHead); err == io.EOF {
-			outOfData = true
-		} else if err != nil {
-			return err
-		} else {
-			eq.safeAttributes = append(eq.safeAttributes, next)
-			return NotEnoughData
-		}
+	newOrigin := eq.prev.Origin()
+	// Check if the L2 unsafe head origin is consistent with the new origin
+	if err := eq.verifyNewL1Origin(ctx, newOrigin); err != nil {
+		return err
+	}
+	eq.origin = newOrigin
+	eq.postProcessSafeL2() // make sure we track the last L2 safe head for every new L1 block
+	if next, err := eq.prev.NextAttributes(ctx, eq.safeHead); err == io.EOF {
+		outOfData = true
+	} else if err != nil {
+		return err
+	} else {
+		eq.safeAttributes = next
```

Comment:
Could you add the `Adding next safe attributes` log (probably just `Storing safe attributes`) here?

Reply:
updated